content/r-package-dev/Documentation.Rmd

```diff
@@ -0,0 +1,253 @@
+          options$fig.path, options$label, options$fig.cur, options$fig.ext, options$fig.cap)
+
+})
+
+opts_chunk$set(
+  echo=TRUE,
+  fig.path="static/doc/",
+  fig.width = 6,
+  fig.height = 6,
+  fig.cap = "TODO"
+)
+
+set.seed(1)
+```
+
+The quality of documentation for an R package can determine it's success or failure. Less documentation generally results in higher burdens for package developer's, so it's best add extensive documentation as early as possible. Consider 3 different audiences when writing pacakge documentation: the users, current or future developers, and future you. The better documented your package the less like you are to be burdened by answering user questions, helping onboard future developers, and spending time remembering details about the code.
+
+## Lesson Objectives
+
+1. Distinguish the different types of documentation for R packages.
+2. Develop documentation for individual functions.
+3. Create a vignette to highlight the top-level package uses.
+4. Edit and update README files.
+
+## Components of package documentation
+
+There are multiple components of package documentation. 
+
+- **Function help files:** Help files are the files that document the inputs and outputs of a function, references and links to related functions, define defaults for the user, and give examples of how to use the function. These files apply to functions at an individual level and are created using Roxygen syntax (we will use the [`roxygen2` package](https://cran.r-project.org/web/packages/roxygen2/vignettes/roxygen2.html).
+
+- **Vignettes:** Packages can have multiple vignettes to illustrate potential workflows or topics within the package. They generally are high-level, illustrating how to apply the package functions to complete an analysis.
+
+- **README:** In the README, you include directions for how to quickly get setup and started. It might include a simple workflow or call out specific functions that people might find interesting.
+
+## Help files
+
+Help files are accessed when users run `?functionName`, e.g. `?mean` brings you to the help file for the function `mean`. Help files describe a function, provide the arguments and argument definitions, show default arguments when applicable, describe what the function returns to the user, points to additional resources or related functions, and provides reproducible examples. Roxygen comments are added to the same file the function is declared. When the package source files are "roxygenized", the roxygen-formatted comments are converted into `.Rd` files and added to the folder `man`. Although, the `.Rd` files look complex, R converts them to human readable formats to show when the `?` or `help` functions are used.
+
+Roxygen style documentation requires the R package `roxygen2`. This package needs to be installed in order to generate the `.Rd` files, but does not need to be listed as a dependency in the `DESCRIPTION` file. **RoxygenNote: 6.0.1 ?!?! in dataRetrieval**
```

Comment:
@ldecicco-USGS what does the `RoxygenNote: 6.0.1` mean in the dataRetrieval description file? Does it need to be included?